apps/datapuller/src/class.ts

```diff
@@ -0,0 +1,67 @@
+import { ClassModel, IClassItem } from "@repo/common";
+import { ClassesAPI } from "@repo/sis-api/classes";
+
+import setup from "./shared";
+import mapClassToNewClass, { CombinedClass } from "./shared/classParser";
+import { Config } from "./shared/config";
+import { fetchActiveTerms, fetchPaginatedData } from "./shared/utils";
+
+async function updateClasses(config: Config) {
+  const log = config.log;
+  const classesAPI = new ClassesAPI();
+
+  log.info("Fetching Active Terms");
+  const activeTerms = await fetchActiveTerms(log, {
+    app_id: config.sis.TERM_APP_ID,
+    app_key: config.sis.TERM_APP_KEY,
+  });
+
+  log.info(activeTerms);
+
+  const classes = await fetchPaginatedData<IClassItem, CombinedClass>(
+    log,
+    classesAPI.v1,
+    activeTerms,
+    "getClassesUsingGet",
+    {
+      app_id: config.sis.CLASS_APP_ID,
+      app_key: config.sis.CLASS_APP_KEY,
+    },
+    (data) => data.apiResponse.response.classes || [],
+    mapClassToNewClass
+  );
+
+  log.info("Example Class:", classes[0]);
+
+  await ClassModel.deleteMany({});
```

Comment:
You should only delete classes you are going to add. This should look like:
```ts
await ClassModel.deleteMany({
  "session.term.id": { "$in": activeTerms }
});
```
(if `activeTerms` is an array of term ids)